deployments/azure-kubernetes-service-helm.yml

```diff
@@ -0,0 +1,124 @@
+# This workflow will build and push an application to a Azure Kubernetes Service (AKS) cluster when you push your code
+#
+# This workflow assumes you have already created the target AKS cluster and have created an Azure Container Registry (ACR)
+# For instructions see:
+#   - https://docs.microsoft.com/en-us/azure/aks/kubernetes-walkthrough-portal
+#   - https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-portal
+#   - https://github.com/Azure/aks-create-action
+#
+# To configure this workflow:
+#
+# 1. Set the following secrets in your repository (instructions for getting these 
+#    https://github.com/Azure/login#configure-a-service-principal-with-a-federated-credential-to-use-oidc-based-authentication):
+#    - AZURE_CLIENT_ID
+#    - AZURE_TENANT_ID
+#    - AZURE_SUBSCRIPTION_ID
+#
+# 2. Set the following environment variables (or replace the values below):
+#    - AZURE_CONTAINER_REGISTRY (name of your container registry / ACR)
+#    - RESOURCE_GROUP (where your cluster is deployed)
+#    - CLUSTER_NAME (name of your AKS cluster)
+#    - CONTAINER_NAME (name of the container image you would like to push up to your ACR)
+#    - SECRET_NAME (name of the secret associated with pulling your ACR image)
+#
+# 3. Choose the appropriate render engine for the bake step https://github.com/Azure/k8s-bake. The config below assumes Helm.
+#    Set your helmChart, overrideFiles, overrides, and helm-version to suit your configuration.
+#    - CHART_PATH (path to your helm chart)
+#    - CHART_OVERRIDE_PATH (path to your helm chart with override values)
+#
+# For more information on GitHub Actions for Azure, refer to https://github.com/Azure/Actions
+# For more samples to get started with GitHub Action workflows to deploy to Azure, refer to https://github.com/Azure/actions-workflow-samples
+# For more options with the actions used below please see the following:
+#   - https://github.com/Azure/login
+#   - https://github.com/Azure/aks-set-context
+#   - https://github.com/Azure/k8s-create-secret
+#   - https://github.com/Azure/k8s-bake
+#   - https://github.com/Azure/k8s-deploy   
+
+name: Build and deploy an app to AKS with Helm
+
+on:
+  push:
+    branches:
+      - $default-branch
+  workflow_dispatch:
+
+env:
+  AZURE_CONTAINER_REGISTRY: "your-azure-container-registry"
+  CONTAINER_NAME: "your-container-name"
+  RESOURCE_GROUP: "your-resource-group"
+  CLUSTER_NAME: "your-cluster-name"
+  IMAGE_PULL_SECRET_NAME: "your-image-pull-secret-name"
+  CHART_PATH: "your-chart-path"
+  CHART_OVERRIDE_PATH: "your-chart-override-path"
+
+permissions:
+  actions: read
+  contents: read
+  id-token: write
+
+jobs:
+  build:
+    permissions:
+      actions: write
+      contents: read
+      id-token: write
+      
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    
+    - name: Azure login
+      uses: azure/login@v1.4.3
```

Comment:
Do you really want to stick to a specific version or using just the major version tag e.g. `v1` is usually a more recommended approach.

Reply:
Using `v1` actually points to the `v1` tag, not the latest version in this case.  I'm not aware of a less specific way to point to the latest version.

<img width="850" alt="Screen Shot 2022-02-22 at 11 54 08 AM" src="https://user-images.githubusercontent.com/18180506/155180220-e28535f4-cfcf-40dd-8404-499a23415db6.png">
<img width="1207" alt="Screen Shot 2022-02-22 at 11 55 57 AM" src="https://user-images.githubusercontent.com/18180506/155180442-d8a223e1-33f7-469b-b04a-37fe92fbfcc9.png">

If you compare the hashes they are the same.

Reply:
AFAIK, `v1` is always supposed to be latest and the action authors usually ensure that. `v1.4.3` may be latest today but will become stale in future.

Reply:
From what we have tested, this does not appear to be the case.